pkg/utils/utils.go

```diff
@@ -74,51 +71,76 @@ func RetrieveVcConfigSecret(params map[string]interface{}, config *rest.Config,
 		logger.WithError(err).Errorf("Failed to get k8s clientset from the given config: %v", config)
 		return err
 	}
+	return retrieveVcConfigSecret(params, clientset, logger)
+}
+
+// Internal method to enable unit testing.
+func retrieveVcConfigSecret(params map[string]interface{}, clientset kubernetes.Interface, logger logrus.FieldLogger) error {
```

Comment:
We don't need a specific func for the unit testing purpose. Please refer to the example I shared below.